src/library_strings.js

```diff
@@ -157,6 +157,9 @@ mergeInto(LibraryManager.library, {
 #if CAN_ADDRESS_2GB
     outIdx >>>= 0;
 #endif
+#if MEMORY64
+  outIdx = Number(outIdx);
```

Comment:
This should not bee needed `outIdx` should already be a number by the time it gets here.  Can you find the backtrace that leads here without `outIdx` being converted to an number already?